clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/Negotiate.java

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+package com.microsoft.aspnet.signalr;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+import java.io.IOException;
+
+public class Negotiate {
+
+    public static NegotiateResponse processNegotiate(String url) throws IOException {
+        return processNegotiate(url, null);
+    }
+
+    public static NegotiateResponse processNegotiate(String url, String accessTokenHeader) throws IOException {
+        url = resolveNegotiateUrl(url);
+        OkHttpClient client = new OkHttpClient();
+        RequestBody body = RequestBody.create(null, new byte[]{});
+        Request.Builder requestBuilder = new Request.Builder()
+                .url(url)
+                .post(body);
+
+        if (accessTokenHeader != null) {
+            requestBuilder.addHeader("Authorization", "Bearer " + accessTokenHeader);
+        }
+
+        Request request = requestBuilder.build();
+
+        Response response = client.newCall(request).execute();
+        String result = response.body().string();
+        return new NegotiateResponse(result);
+    }
+
+    private static String resolveNegotiateUrl(String url) {
```

Comment:
Add some tests for this please. See https://github.com/aspnet/SignalR/blob/release/2.2/clients/ts/signalr/tests/Common.ts#L17 and https://github.com/aspnet/SignalR/blob/release/2.2/clients/ts/signalr/tests/HttpConnection.test.ts#L220 for some reference

Reply:
Good point. 👍